services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -182,6 +182,19 @@ def _update_resource_limits_and_reservations(
         spec["environment"] = environment
 
 
+_METRICS_PREFIX: Final[str] = "io.simcore.container"
+
+
+def _update_container_labels(service_spec: ComposeSpecLabel, user_id: UserID) -> None:
+    for spec in service_spec["services"].values():
+        labels = spec.get("labels", [])
+
+        # this labels is primarily used for metrics scraping
+        labels.append(f"{_METRICS_PREFIX}.user.id={user_id}")
```

Comment:
BUG: avoid repeated labels !!! 
Remember that we had this bug before with la constraint for ``linux`` nodes!